lib/net/ssh/transport/session.rb

```diff
@@ -61,12 +61,12 @@ def initialize(host, options={})
       @bind_address = options[:bind_address] || nil
       @options = options
 
-      debug { "establishing connection to #{@host}:#{@port}" }
-
       @socket =
         if (factory = options[:proxy])
+          debug { "establishing connection to proxy" }
```

Comment:
It is completely ignored. When connecting through a proxy, you only have one connection: the proxy. Once you have established a proxy connection and make a request akin to `ssh -W host:port proxy.host.name` the proxy connects on your behalf to *something* which might (or might not) be host:port.

In any case, the only connection the client makes is to a proxy. Debugging here should report connecting to a proxy because that's the connection being made. It's also the connection that will fail in this context. We need to inform the user that it's the connection to the proxy that failed, not the connection to the final endpoint.